src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,220 @@
+                        Guid.NewGuid(),
+                        prefix: prefix,
+                        capitalizationScheme: Capitalization.CamelCase),
+                    DiagnosticSeverity.Hidden));
+        }
+
+        private string GenerateFieldName(
+            IPropertySymbol property, 
+            ImmutableArray<NamingRule> rules)
+        {
+            var propertyName = property.Name;
+            var fieldName = "";
+            var isStatic = property.IsStatic;
+            foreach (var rule in rules)
+            {
+                if (rule.SymbolSpecification.AppliesTo(
+                    new SymbolKindOrTypeKind(SymbolKind.Field), 
+                    isStatic ? DeclarationModifiers.Static : DeclarationModifiers.None, 
+                    Accessibility.Private))
+                {
+                    fieldName = rule.NamingStyle.MakeCompliant(propertyName).Single();
+                    break;
+                }
+            }
+
+            return GetUniqueName(fieldName, property);
+        }
+
+        private async Task<SyntaxNode> ExpandPropertyAndAddFieldAsync(
+            Document document,
+            SyntaxNode property,
+            SyntaxNode root,
+            IPropertySymbol propertySymbol,
+            string fieldName,
+            SyntaxGenerator generator,
+            CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+
+            // Create full property. If the auto property had an initial value
+            // we need to remove it and later add it to the backing field
+            var accessorTuple = await GetNewAccessorsAsync(
+                document,
+                property,
+                fieldName,
+                generator,
+                cancellationToken)
+                    .ConfigureAwait(false);
+
+            var fullProperty = generator
+                    .WithAccessorDeclarations(GetPropertyWithoutInitializer(property), accessorTuple.newSetAccessor == null ? new SyntaxNode[] { accessorTuple.newGetAccessor } : new SyntaxNode[] { accessorTuple.newGetAccessor, accessorTuple.newSetAccessor })
+                    .WithLeadingTrivia(property.GetLeadingTrivia())
+                    .WithAdditionalAnnotations(Formatter.Annotation);
+            var editor = new SyntaxEditor(root, workspace);
+            editor.ReplaceNode(property, fullProperty);
+
+            // add backing field, plus initializer if it exists 
+            var newField = CodeGenerationSymbolFactory.CreateFieldSymbol(default, Accessibility.Private, DeclarationModifiers.From(propertySymbol), propertySymbol.Type, fieldName, initializer: GetInitializerValue(property));
+            var containingType = GetTypeBlock(propertySymbol.ContainingType.DeclaringSyntaxReferences.FirstOrDefault().GetSyntax(cancellationToken));
+
+            editor.ReplaceNode(containingType, (currentTypeDecl, _) =>
+                {
```

Comment:
no need for the body or 'return', this is short enough to just have a simple lambda.